x-pack/elastic-agent/pkg/agent/operation/monitoring.go

```diff
@@ -339,6 +339,16 @@ func (o *Operator) getMonitoringFilebeatConfig(outputType string, output interfa
 		"output": map[string]interface{}{
 			outputType: output,
 		},
+		"processors": []map[string]interface{}{
+			{
+				"add_fields": map[string]interface{}{
+					"target": "agent",
+					"fields": map[string]interface{}{
+						"id": o.agentInfo.AgentID(),
```

Comment:
out of curiosity, why is this processor not part of the input processors above?

Reply:
It could go there, but by putting here it should be automatically applied to every input. In effect the processor is appended to the list of each input's processors. This way just means less code duplication.

Reply:
Apparently in the config transformation the root `processors` is dropped or not carried over to the final config so that didn't work. I opened #26776 to move add_fields into each input.